File: Annadata/app/src/main/java/com/mananhirak/annadata/activitys/FoodBuy.java
package com.mananhirak.annadata.activitys;

import android.app.Dialog;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.ViewPager2;

import com.airbnb.lottie.LottieAnimationView;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.mananhirak.annadata.R;
import com.mananhirak.annadata.foodcycle.FRAGMENT_ADAPTER;

import java.util.Objects;

public class FoodBuy extends AppCompatActivity implements FirebaseAuth.AuthStateListener {


    ImageView ivBack;
    String SellerIdBySharableLink;
    TabLayout tabLayout;
    ViewPager2 viewPager2;
    FRAGMENT_ADAPTER fragmentAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_food_buy);
        Objects.requireNonNull(getSupportActionBar()).hide();

        IdSaver();

        ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });

    }


    @Override
    public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
        if(FirebaseAuth.getInstance().getCurrentUser()==null) {
            Intent intent = new Intent(this, MainActivity.class);
            startActivity(intent);
            finish();
        }else{
            MarketCycle(Objects.requireNonNull(firebaseAuth.getCurrentUser()));
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        FirebaseAuth.getInstance().addAuthStateListener(this);
    }

    @Override
    protected void onStop() {
        super.onStop();
        FirebaseAuth.getInstance().removeAuthStateListener(this);
    }

    private void MarketCycle(FirebaseUser user){

        fragmentAdapter = new FRAGMENT_ADAPTER(this);
        Intent get_intent=getIntent();
        SellerIdBySharableLink=get_intent.getStringExtra("SellerIDFromSharableLink");
        if(SellerIdBySharableLink!=null) {

            fragmentAdapter.TiffinLinkSellerId=SellerIdBySharableLink;
        }

        SetUpCards();

    }
    public  void SetUpCards(){

        viewPager2.setAdapter(fragmentAdapter);

        new TabLayoutMediator(tabLayout, viewPager2,
                new TabLayoutMediator.TabConfigurationStrategy() {
                    @Override
                    public void onConfigureTab(@NonNull TabLayout.Tab tab, int position) {
                        if(position==0) tab.setText("Market");
                        else tab.setText("Personal");
                    }
                }).attach();
    }

    private void IdSaver(){
        ivBack = findViewById(R.id.ivBack);
        tabLayout= findViewById(R.id.act_fd_tablayout);
        viewPager2=findViewById(R.id.act_fd_viewpager2);
    }

}

File: Annadata/app/src/main/java/com/manan/annadata/activitys/FoodBuy.kt
package com.manan.annadata.activitys

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
import com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseAuth.AuthStateListener
import com.google.firebase.auth.FirebaseUser
import com.manan.annadata.R
import com.manan.annadata.foodcycle.FRAGMENT_ADAPTER
import java.util.*

class FoodBuy constructor() : AppCompatActivity(), AuthStateListener {
    var ivBack: ImageView? = null
    var SellerIdBySharableLink: String? = null
    var tabLayout: TabLayout? = null
    var viewPager2: ViewPager2? = null
    var fragmentAdapter: FRAGMENT_ADAPTER? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_food_buy)
        Objects.requireNonNull(getSupportActionBar())?.hide()
        IdSaver()
        ivBack!!.setOnClickListener(object : View.OnClickListener {
            public override fun onClick(view: View) {
                finish()
            }
        })
    }

    public override fun onAuthStateChanged(firebaseAuth: FirebaseAuth) {
        if (FirebaseAuth.getInstance().getCurrentUser() == null) {
            val intent: Intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
            finish()
        } else {
            firebaseAuth.currentUser?.let { MarketCycle(it) }
        }
    }

    override fun onStart() {
        super.onStart()
        FirebaseAuth.getInstance().addAuthStateListener(this)
    }

    override fun onStop() {
        super.onStop()
        FirebaseAuth.getInstance().removeAuthStateListener(this)
    }

    private fun MarketCycle(user: FirebaseUser) {
        fragmentAdapter = FRAGMENT_ADAPTER(this)
        val get_intent: Intent = getIntent()
        SellerIdBySharableLink = get_intent.getStringExtra("SellerIDFromSharableLink")
        if (SellerIdBySharableLink != null) {
            fragmentAdapter!!.TiffinLinkSellerId = SellerIdBySharableLink
        }
        SetUpCards()
    }

    fun SetUpCards() {
        viewPager2!!.setAdapter(fragmentAdapter)
        TabLayoutMediator(
            (tabLayout)!!, (viewPager2)!!,
            object : TabConfigurationStrategy {
                public override fun onConfigureTab(tab: TabLayout.Tab, position: Int) {
                    if (position == 0) tab.setText("Market") else tab.setText("Personal")
                }
            }).attach()
    }

    private fun IdSaver() {
        ivBack = findViewById(R.id.ivBack)
        tabLayout = findViewById(R.id.act_fd_tablayout)
        viewPager2 = findViewById(R.id.act_fd_viewpager2)
    }
}